geode-core/src/test/java/org/apache/geode/internal/cache/DiskStoreImplIntegrationTest.java

```diff
@@ -80,12 +89,53 @@ public void cleansUpOrphanedBackupFilesOnDiskStoreCreation() throws Exception {
     tempDirs.forEach(tempDir -> assertThat(Files.exists(tempDir)).isFalse());
   }
 
+  @Test
+  public void queueSizeStatIncrementedAfterAsyncFlush() throws Exception {
+    File baseDir = temporaryDirectory.newFolder();
+    final int QUEUE_SIZE = 50;
+    createRegionWithDiskStoreAndAsyncQueue(baseDir, QUEUE_SIZE);
+    Awaitility.await().atMost(1, TimeUnit.MINUTES).until(() -> diskStoreStats.getQueueSize() == 0);
+
+    putEntries(QUEUE_SIZE - 1);
+    Awaitility.await().atMost(1, TimeUnit.MINUTES)
+        .until(() -> diskStoreStats.getQueueSize() == QUEUE_SIZE - 1);
+
+    putEntries(1);
+    Awaitility.await().atMost(1, TimeUnit.MINUTES).until(() -> diskStoreStats.getQueueSize() == 0);
+  }
+
+  private void putEntries(int numToPut) {
+    for (int i = 1; i <= numToPut; i++) {
+      aRegion.put(i, i);
+    }
+  }
+
   private void createRegionWithDiskStore(File baseDir) {
     cache.createDiskStoreFactory().setDiskDirs(new File[] {baseDir}).create(DISK_STORE_NAME);
     cache.<String, String>createRegionFactory(RegionShortcut.PARTITION_PERSISTENT)
         .setDiskStoreName(DISK_STORE_NAME).create(REGION_NAME);
   }
 
+  private void createRegionWithDiskStoreAndAsyncQueue(File baseDir, int queueSize) {
+    createDiskStoreWithQueue(baseDir, queueSize);
+
+    RegionFactory regionFactory =
+        cache.<String, String>createRegionFactory(RegionShortcut.PARTITION_PERSISTENT);
+    regionFactory.setDiskSynchronous(false);
+    regionFactory.setDiskStoreName(DISK_STORE_NAME);
+    aRegion = regionFactory.create(REGION_NAME);
+  }
+
+  private void createDiskStoreWithQueue(File baseDir, int queueSize) {
+    final int TIME_INTERVAL = 300000;
```

Comment:
just as you pass in the queueSize, I'd also pass in the timeInterval instead of hard coding it.
That way in you unit test method queueSizeStatIncrementedAfterAsyncFlush, you can have a large TIME_INTERVAL in it that you pass in. I think that will make that test easier to understand when reading that test method. When I was reviewing it, I was worried we might have a race in it when the flush happens because of time. I had to get all the done to this point to realize that it was not an issue because of the large TIME_INTERVAL